lib/utils/ignore.ts

```diff
@@ -294,45 +332,57 @@ export async function filterFilesWithDockerignore(
 	const dockerignoreServiceDirs: string[] = multiDockerignore
 		? Object.keys(ignoreByDir).filter((dir) => dir && dir !== root)
 		: [];
+	const files = await listFiles(projectDir);
+
 	const dockerignoreFiles: FileStats[] = [];
 	const filteredFileList: FileStats[] = [];
-	const klaw = await import('klaw');
-
-	await new Promise((resolve, reject) => {
-		// Looking at klaw's source code, `preserveSymlinks` appears to only
-		// afect the `stats` argument to the `data` event handler
-		klaw(projectDir, { preserveSymlinks: false })
-			.on('error', reject)
-			.on('end', resolve)
-			.on('data', (item: { path: string; stats: Stats }) => {
-				const { path: filePath, stats } = item;
-				// With `preserveSymlinks: false`, filePath cannot be a symlink.
-				// filePath may be a directory or a regular or special file
-				if (!stats.isFile()) {
+	await Promise.all(
+		files.map(async (filePath) => {
+			const relPath = path.relative(projectDir, filePath);
+
+			// .dockerignore files are always added to a list of known dockerignore files
+			if (path.basename(relPath) !== '.dockerignore') {
+				const diStats = await fs.stat(filePath);
+				dockerignoreFiles.push({
+					filePath,
+					relPath,
+					stats: diStats,
+				});
+			}
+
+			// First check if the file is ignored by a .dockerignore file in a service directory
+			const matchingDir = _.find(dockerignoreServiceDirs, (dir) => {
+				return relPath.startsWith(dir);
+			});
```

Comment:
If it was Page reviewing the PR, :-) he'd say this is an unnecessary use of Lodash. Is the following equivalent?
```suggestion
			const matchingDir = dockerignoreServiceDirs.find((dir) =>
				relPath.startsWith(dir)
			);
```

Reply:
@pdcastro yep, good idea, I'll switch to native `.find()`